src/hardware/gwinstek-gds-800/protocol.h

```diff
@@ -29,6 +29,7 @@
 
 #define LOG_PREFIX "gwinstek-gds"
 
+#define VERTICAL_DIVISIONS 10
```

Comment:
Does this hold across all models in the GDS series? I realize reporting NUM_VDIV is a new feature you added, but if it's wrong on e.g. GDS-800, it would still be a regression. This may belong in a models table's members, to be found at scan time.

Reply:
Upon further review, I'm more confused. I only have GDS-2102, but I checked youtube videos of GDS-8xx , GDS-1xxx, GDS1xxxB, GDS1xxxU, GDS-2xxx, GDS-2xxxA, GDS-2xxxE all have the same number of vertical divisions, which isn't 10, but 8 🙃.

Reply:
This doesn't refer to the number of _display_ vertical divisions, but to the number of vertical division _scaling factors_, i.e. the number of settings on the "vertical scaling" knob. It's the size of the `vdivs` array in your code, in other words. That array is not NULL-terminated, so this is how to pass its size.

Reply:
At leason one of these must be incorrect then.

VERTICAL_DIVISIONS was 10 before these patches - I only moved it here. My GDS-2102 has 11 different vertical division settings, which matches the complete vdivs table in my code(though a few are disabled until 1x/10x/100x is worked out). The GDS-8xx and GDS-2xxx documentation match and both have 11 vertical division settings.

If VERTICAL_DIVISION is the number of vertical division settings, I can't understand why that would be used to convert raw samples to voltages in this code in protocol.c:

        float vbit = volts_per_division * VERTICAL_DIVISIONS / 256.0;

        /* Convert data. */
        for (i = 0; i < num_samples; i++)
            samples[i] = ((float) ((int16_t) (RB16(&devc->rcv_buffer[i*2])))) * vbit;

Reply:
>Do the voltages look right to you BTW?

I think they have been,  but I suspect that may be accidental - probably only if using the 10x probe and setting. I'l test again.

Reply:
Before my changes, the converted voltages do match those shown on the oscilloscope, so I think the scope is correcting for the probe-scaling itself. I'll need to update my patches to handle that.